hack/capi-sync-provider.sh

```
@@ -0,0 +1,117 @@
+            --rewrite "sigs.k8s.io/cluster-api-provider-openstack/api/=github.com/openshift/hypershift/pkg/capi/openstack/" \
+            --rewrite "sigs.k8s.io/cluster-api-provider-openstack/pkg/utils/=github.com/openshift/hypershift/pkg/capi/openstack/utils/"
+        ;;
+    *)
+        echo "ERROR: unknown provider: $PROVIDER" >&2
+        exit 1
+        ;;
+esac
+
+# Remove conversion machinery — HyperShift uses the copied types for CRD
+# generation and controller references, not for CAPI's conversion webhooks.
+find "$DST_DIR" -name '*conversion*' -print | xargs rm -f
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)